packages/grid/x-data-grid/src/components/GridRow.tsx

```diff
@@ -125,10 +128,39 @@ function GridRow(props: React.HTMLAttributes<HTMLDivElement> & GridRowProps) {
     classes: rootProps.classes,
     editing: apiRef.current.getRowMode(rowId) === GridRowModes.Edit,
     editable: rootProps.editMode === GridEditModes.Row,
+    rowHeight,
   };
 
   const classes = useUtilityClasses(ownerState);
 
+  React.useLayoutEffect(() => {
+    if (rowHeight === 'auto' && ref.current && typeof ResizeObserver === 'undefined') {
+      // Fallback for IE
+      apiRef.current.unstable_storeRowHeightMeasurement(rowId, ref.current.clientHeight);
+    }
+  });
+
+  React.useLayoutEffect(() => {
+    if (currentPage.range) {
+      apiRef.current.unstable_setLastMeasuredRowIndex(index - currentPage.range.firstRowIndex);
```

Comment:
I don't get why is it a `-`, and not a `+` in `index - currentPage.range.firstRowIndex`

Reply:
`currentPage.range.firstRowIndex` is added to the index below and I need to remove it here to make the index relative to only the visible rows in the current page.

https://github.com/mui/mui-x/blob/afd72a9c304a1608d2b325df761bdefe4573f878/packages/grid/x-data-grid/src/hooks/features/virtualization/useGridVirtualScroller.tsx#L388

The last measured index is supposed to always begin at 0 in every page.

Reply:
Ok, maybe a comment, or `const rowIndexInCurrentPage = index - currentPage.range.firstRowIndex` would help futur readers

Reply:
Done.